lib/checkcondition.cpp

```diff
@@ -903,6 +903,108 @@ static int countPar(const Token *tok1, const Token *tok2)
     return par;
 }
 
+// Clarify '[if|while|function] (x = foo() && bar())'
+void CheckCondition::clarifyAssignmentAndLogicalCondition()
+{
+    if (!_settings->isEnabled("style"))
+        return;
+
+    const SymbolDatabase *symbolDatabase = _tokenizer->getSymbolDatabase();
+    const std::size_t functions = symbolDatabase->functionScopes.size();
+    for (std::size_t i = 0; i < functions; ++i) {
+        const Scope * scope = symbolDatabase->functionScopes[i];
+        for (const Token* tok = scope->classStart->next(); tok != scope->classEnd; tok = tok->next()) {
+
+            const Token* root = tok;
+            if(!tok || tok->str() != "(") {
```

Comment:
OK, in this case, it's maybe redundant but further in the code, astOperand2() can return a NULL pointer (while testing TestCondition::oppositeInnerCondition for example).
Regarding the pattern [ ! ptr || ptr->f()], if ptr could be NULL, !ptr would be true and the rest of the condition would not be evaluated (if (a || b) with a == true, b is not evaluated because, a || b is always true, whatever is the value of b).
So, if you're sure that Token::next() will never return NULL, I remove !tok but I let it when tok comes from astOperand().